### PR TITLE
Build db models

### LINKED ISSUE #7072: [CVEs] Setup models and schema
## Summary

Build models and schema for CVEs to create tables.

### PR DESCRIPTION
**Pull request by carkod**
## Done

- Models and schema (no API yet)

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8001/
- No QA really, just want to make sure the models are correct and I am creating the relationships correctly because, once this is run, we'll start messing the db if it is not correct.

## Issue / Card

Fixes #7072

### CODE REVIEW

File: webapp/security/models.py [hunk 1]
```diff
@@ -37,17 +37,54 @@
...
+    references = relationship("CVEReference", secondary=cve_references)
+    bugs = relationship("Bug", secondary=cve_bugs)
+    packages = relationship("Package", secondary=package_release_status)
 
 
 class Notice(Base):
     __tablename__ = "notice"
 
-    id = Column(String, primary_key=True)
+    id = Column(Integer, primary_key=True)
```
Comment (jpmartinspt) [inline webapp/security/models.py]: The id of a notice is a `String`

File: webapp/app.py [hunk 2]
```diff
@@ -146,6 +148,11 @@ def utility_processor():
 app.add_url_rule("/security/notices/<notice_id>", view_func=notice)
 
 
+# cve section
+app.add_url_rule("/security/cve", view_func=cve_index)
+app.add_url_rule("/security/<regex('cve-d{4}-d{4,7}'):cve_id>", view_func=cve)
```
Comment (jpmartinspt) [inline webapp/app.py]: why not just `security/cve/<cve_id>`

File: webapp/security/models.py (hunk 1 above)
Comment (carkod) [reply webapp/security/models.py]: OK, I thought that was a mistake, because the others are all Integers?

Comment (jpmartinspt) [reply webapp/security/models.py]: Nop it should be a String

File: webapp/app.py (hunk 2 above)
Comment (carkod) [reply webapp/app.py]: It was requested by [design]
(https://docs.google.com/document/d/1VuAElUpFay7MBFNmE3NhSyY9TkXiqHcA1ZHQ11AYO-Q/edit#heading=h.ua7qthpl074)

File: webapp/security/models.py [hunk 3]
```diff
@@ -37,11 +60,50 @@
...
+    notes = Column(String)
+    mitigation = Column(String)
+    priority = Column(String)
+    discovered_by = Column(String)
+    assigned_to = Column(String)
+    approved_by = Column(String)
+    cvss = Column(String)  # CVSS 3 and Base Score
+    references = relationship("CVEReference", secondary=cve_references)
+    bugs = relationship("Bug", secondary=cve_bugs)
+    packages = relationship("Package", secondary=package_release_status)
```
Comment (jpmartinspt) [inline webapp/security/models.py]: I think what you want here is a `secondary=cve_packages` here(You need to create that table).

Then you probably want to add `releases_status = relationship("Release", secondary=package_release_status)` to the Packages models.

Does that make sense?

File: webapp/security/models.py [hunk 4]
```diff
@@ -37,11 +60,50 @@
...
+    "cve_references",
+    Base.metadata,
+    Column("cve_id", String, ForeignKey("cve.id")),
+    Column("cve_reference_id", Integer, ForeignKey("cve_reference.id")),
+)
+
+package_release_status = Table(
+    "package_release_status",
+    Base.metadata,
+    Column("package_id", String, ForeignKey("package.id")),
```
Comment (jpmartinspt) [inline webapp/security/models.py]: This should be an `Integer`

File: webapp/security/models.py (hunk 3 above)
Comment (carkod) [reply webapp/security/models.py]: Ok yeah. So adding that relationship will link it to the existent releases table?

Comment (jpmartinspt) [reply webapp/security/models.py]: Yep

File: webapp/security/models.py [hunk 5]
```diff
@@ -7,13 +7,36 @@
     JSON,
     String,
     Table,
+    Enum,
 )
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm import relationship
-
+import enum
```
Comment (jpmartinspt) [inline webapp/security/models.py]: stdlib imports at the top please.

File: .env [hunk 6]
```diff
@@ -2,7 +2,7 @@ PORT=8001
 FLASK_DEBUG=true
 DEVEL=true
 ADVANTAGE_API=https://contracts.canonical.com/
-DATABASE_URL=sqlite:///usn.sqlite3
+DATABASE_URL=sqlite:///
```
Comment (jpmartinspt) [inline .env]: Any reason we are changing this? is quite useful to persist on disk so we have the same data in between re-runs.

Comment (carkod) [reply .env]: Ok, I thought this was the right way to do it. So should i keep it as `sqlite:///usn.sqlite3`?

File: webapp/security/models.py [hunk 7]
```diff
@@ -37,11 +61,59 @@
...
+    Column("status", Enum(PackageStatus)),
+    extend_existing=True,
+)
+
 
 class CVE(Base):
     __tablename__ = "cve"
 
-    id = Column(String, primary_key=True)
+    id = Column(String, primary_key=True, unique=True)
```
Comment (jpmartinspt) [inline webapp/security/models.py]: I believe `unique=True` is not necessary here case the `primary_key=True` constraint already enforces uniqueness.

Comment (carkod) [reply webapp/security/models.py]: ok, didn't know that

**Comment by jpmartinspt** [comment]
This needed a migration.

**Pull request merged**